### PR TITLE
Fix UTM to Lat Lon. 

### LINKED ISSUE #12: Missing method Node from keyvalue pairs
Not sure why but creating the Node, Way and Relation from KeyValues are missing in the master

### PR DESCRIPTION
**Pull request by rolyhudson**
Closes #12 #4 and #18

Fixes to UTM to Lat Lon issues. Add missing create methods for node needed for this fix and convert longitude to utm zone.

### Test files
in the repo examples folder:
Open ConvertCoords.gh  to test the fix to #18 
Open M25.gh to test the fix to #4

### CODE REVIEW

**Comment by rolyhudson** [comment]
Not to disregard your review @emidio-piermarini just noting that we need a test file to verify the change for fixing #4 and improving the naming for clarity of convert methods.

File: OpenStreetMap_Engine/Convert/ToUTMZone.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline OpenStreetMap_Engine/Convert/ToUTMZone.cs:3]: Copyright should be 2020

File: OpenStreetMap_Engine/Convert/ToUTMZone.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        /***************************************************/
+        /****           Public Methods                  ****/
+        /***************************************************/
+        public static int ToUTMZone(double longitude)
+        {
+            return (int)Math.Ceiling((longitude + 180) / 6);
+        }
+        /***************************************************/
+        public static int ToUTMZone(this Node node)
```
Comment (FraserGreenroyd) [inline OpenStreetMap_Engine/Convert/ToUTMZone.cs:38]: Would be good to add blank lines around the breakers to aid readability

File: OpenStreetMap_Engine/Create/Node.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline OpenStreetMap_Engine/Create/Node.cs:3]: Should be 2020

File: OpenStreetMap_Engine/Create/Node.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+        }
+        /***************************************************/
+        public static Node Node(Dictionary<string, string> keyvalues)
+        {
+            return new Node()
+            {
+                KeyValues = keyvalues
+            };
+        }
```
Comment (FraserGreenroyd) [inline OpenStreetMap_Engine/Create/Node.cs:48]: Is this method needed given the BHoM_UI gives us default create methods which would handle this?

Comment (rolyhudson) [reply OpenStreetMap_Engine/Create/Node.cs:48]: Good point and a reminder of an outstanding issue I raised. Its a work around: https://github.com/BHoM/Grasshopper_Toolkit/issues/495. 
Which I checked again today and is still an issue:

The osm ids must be int64 but the default create methods don't see to like int64 but want dates instead 😒

File: OpenStreetMap_Engine/Convert/ToUTMZone.cs (hunk 2 above)
Comment (rolyhudson) [reply OpenStreetMap_Engine/Convert/ToUTMZone.cs:38]: Done but I'm not sure I agree as we end up with more scrolling which I think can be argued to be more damaging to readability.  I ❤️ talking whitespace with you @FraserGreenroyd

File: OpenStreetMap_Engine/Create/Node.cs (hunk 4 above)
Comment (FraserGreenroyd) [reply OpenStreetMap_Engine/Create/Node.cs:48]: That makes sense, seems valid to stay :smile:

**Comment by rolyhudson** [comment]
Trying to fix the "This branch cannot be rebased due to conflicts"...

**Pull request closed without merging**